scene/gui/text_edit.cpp

```diff
@@ -2246,7 +2246,7 @@ void TextEdit::_gui_input(const Ref<InputEvent> &p_gui_input) {
 					selection.selecting_column = col;
 				}
 
-				if (!mb->is_double_click() && (OS::get_singleton()->get_ticks_msec() - last_dblclk) < 600 && cursor.line == prev_line) {
+				if (!mb->is_double_click() && (OS::get_singleton()->get_ticks_msec() - last_dblclk) < 600 && mb->get_position().distance_to(last_dblclk_pos) < 5) {
```

Comment:
I would probably move the tolerance into some kind of constant (local `const int` is fine for this). This would avoid using a "magic number" of sorts.

Reply:
600 is a magic number too.

Reply:
I actually had it as a constant like this:
```c++
const int triple_click_tolerance = 5;
```
But I changed it because there's other constants like that too. Should I make constants for both the numbers then? eg:
```c++
const int triple_click_timeout = 600;
const int triple_click_tolerance = 5;
```

Reply:
> Should I make constants for both the numbers then?

Sure :slightly_smiling_face: